### PR TITLE
Make `common_isotopes`, `known_isotopes`, and `stable_isotopes` each return a `ParticleList`

### PR DESCRIPTION
**Pull request by namurphy**
Whilst working on #2458, I noticed that a few of the functions in `plasmapy.particles.atomic` each returned a `list` containing `Particle` objects.  This PR changes it so that they return a `ParticleList`.

### CODE REVIEW

File: plasmapy/particles/atomic.py [hunk 1]
```diff
@@ -664,11 +664,11 @@ def common_isotopes_for_element(
         CommonIsotopes = [
             isotope
             for isotope in isotopes
-            if "abundance" in _isotopes.data_about_isotopes[isotope]
+            if "abundance" in _isotopes.data_about_isotopes[isotope.isotope]
```
Comment (namurphy) [inline plasmapy/particles/atomic.py:667]: `Particle.isotope` returns a string with the symbol of the isotope, which is expected by that `dict`.

File: plasmapy/particles/atomic.py [hunk 2]
```diff
@@ -783,7 +783,7 @@ def stable_isotopes_for_element(
         return [
             isotope
             for isotope in KnownIsotopes
-            if _isotopes.data_about_isotopes[isotope]["stable"] == stable_only
+            if _isotopes.data_about_isotopes[isotope.symbol]["stable"] == stable_only
```
Comment (namurphy) [inline plasmapy/particles/atomic.py:786]:
```suggestion
            if _isotopes.data_about_isotopes[isotope.isotope]["stable"] == stable_only
```

File: plasmapy/particles/tests/test_atomic.py [hunk 3]
```diff
@@ -483,18 +483,14 @@ def test_isotopic_abundance() -> None:
         isotopic_abundance("Og-2")
 
 
-isotopic_abundance_elements = (
-    atomic_number(atomic_numb) for atomic_numb in range(1, 119)
-)
```
Comment (namurphy) [inline plasmapy/particles/tests/test_atomic.py:488]: I seem to have discovered Python generators in Dec 2017 and perhaps maybe had just a little bit too much fun with them...

File: plasmapy/particles/tests/test_atomic.py
Comment (namurphy) [inline plasmapy/particles/tests/test_atomic.py:1]: This file has some incidental refactorings that I made while trying to debug a thing.

File: plasmapy/particles/atomic.py (hunk 1 above)
Comment (ejohnson-96) [reply plasmapy/particles/atomic.py:667]: I thought we didn't return dict objects?

File: plasmapy/particles/atomic.py (hunk 2 above)
Comment (ejohnson-96) [reply plasmapy/particles/atomic.py:786]: Updated to isotope

**Review by ejohnson-96 — APPROVED** [review]
👍

File: plasmapy/particles/atomic.py (hunk 1 above)
Comment (namurphy) [reply plasmapy/particles/atomic.py:667]: The dict here was a private one containing isotope data, so it wasn't part of our public API.  I'd say that there are times when a dict is a good choice for a return value, and other times when it's worth creating something like a class or namedtuple.  As with everything, there are tradeoffs!

**Comment by namurphy** [comment]
pre-commit.ci autofix

**Pull request merged**